cdap-ui/app/cdap/components/PipelineDetails/PipelineRuntimeArgsCounter/index.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import Proptypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+import { getFilteredRuntimeArgs } from 'components/PipelineConfigurations/Store/ActionCreator';
+require('./PipelineRuntimeArgsCounter.scss');
+
+const getTotalRuntimeArgsCount = (runtimeArgs) => {
+  let count = 0;
+  runtimeArgs.pairs.forEach(runtimeArg => {
+    if (!isEmpty(runtimeArg.key)) {
```

Comment:
should we check if the value is empty as well?

Reply:
I had it the way initially but when we have macros, which won't have a value when you load it the first time, the user would expect that they have 2 runtime arguments but with no value. So if there is a valid key I considered it as a runtime argument